sentry-core/src/main/java/io/sentry/core/transport/HttpTransport.java

```diff
@@ -297,46 +277,83 @@ public boolean isRetryAfter(final @NotNull String itemType) {
     int responseCode = -1;
 
     try (final OutputStream outputStream = connection.getOutputStream();
-        final Writer writer = new BufferedWriter(new OutputStreamWriter(outputStream, UTF_8))) {
+        final GZIPOutputStream gzip = new GZIPOutputStream(outputStream);
+        final Writer writer = new BufferedWriter(new OutputStreamWriter(gzip, UTF_8))) {
```

Comment:
`m`: The whole function here almost looks the same as `send(SentryEvent)`. Can we maybe refactor this so we don't have it duplicated?

Reply:
@philipphofmann not really, because we use `try-with-resources` and to be able to use it, we can't use a method that returns all the types and are free'd afterwards

Reply:
If the method returns the `writer` the `try-with-resources` is calling `close()` on the `writer` and the `writer` calls close on the `gzip` and `gzip` calls close on the `outputStream`. I think this should work fine.

Reply:
it doesn't, they only guarantee to be closed if they are inside of the try-with-resources.
if creating `BufferedWriter` throws, only the `Writer` will be freed and the previous `OutputStream` and `GZIPOutputStream` won't

Reply:
We could wrap the creation of the `BufferedWriter` in the extracted method in a try catch and then we would be good. Nevermind it is not so important to me. 
What is important to me is that `public @NotNull TransportResult send(final @NotNull SentryEnvelope envelope) throws IOException` and `public @NotNull TransportResult send(final @NotNull SentryEvent event) throws IOException` are almost identical. The only difference is the log message and that sending the envelope has an extra `catch (Exception e)`.

Reply:
@philipphofmann the extra `catch` is due to the nature of serializing an envelope which is not only GSON's job but also us parsing it correctly, it may throw Exception due to invalid/missing data or something like.
if you don't use `try-with-resources` and wrap to normal try/catch altogether, you'll have way more code to close and end resources altogether, it was like this before, and try with resources is making sure that none of them leak even if it throws, the downside is a bit of duplicated code, but is probably gonna be worked on in the future as the `/store` endpoint will be deprecated and one of this method is gonna be removed anyway, so I'd not focus on refactoring this if that's ok.

Reply:
It is fine for me to keep it like it is.